cmd/dep/root_analyzer.go

```diff
@@ -0,0 +1,550 @@
+	importers := []importer{
+		newGlideImporter(a.ctx, a.sm),
+	}
+
+	for _, i := range importers {
+		if i.HasConfig(dir) {
+			return i.Import(dir, pr)
+		}
+	}
+
+	return nil, nil, nil
+}
+
+// Perform analysis of the filesystem tree rooted at path, with the
+// root import path importRoot, to determine the project's constraints, as
+// indicated by a Manifest and Lock.
+func (a *rootAnalyzer) deriveManifestAndLockFromGopath() (*dep.Manifest, *dep.Lock, error) {
+	var err error
+
+	a.ctx.Loggers.Err.Println("Searching GOPATH for projects...")
+	a.pd, err = getProjectData(a.ctx, a.pkgT, a.sm)
+	if err != nil {
+		return nil, nil, err
+	}
+	m := &dep.Manifest{
+		Dependencies: a.pd.constraints,
+	}
+
+	// Make an initial lock from what knowledge we've collected about the
+	// versions on disk
+	l := &dep.Lock{
+		P: make([]gps.LockedProject, 0, len(a.pd.ondisk)),
+	}
+
+	for pr, v := range a.pd.ondisk {
+		// That we have to chop off these path prefixes is a symptom of
+		// a problem in gps itself
+		pkgs := make([]string, 0, len(a.pd.dependencies[pr]))
+		prslash := string(pr) + "/"
+		for _, pkg := range a.pd.dependencies[pr] {
+			if pkg == string(pr) {
+				pkgs = append(pkgs, ".")
+			} else {
+				pkgs = append(pkgs, trimPathPrefix(pkg, prslash))
+			}
+		}
+
+		l.P = append(l.P, gps.NewLockedProject(
+			gps.ProjectIdentifier{ProjectRoot: pr}, v, pkgs),
+		)
+	}
+
+	// Copy lock before solving. Use this to separate new lock projects from soln
+	a.origL = l
+
+	return m, l, nil
+}
+
+// DeriveManifestAndLock evaluates a dependency for existing dependency manager configuration (ours or external) and passes the configuration found back to the solver
+func (a *rootAnalyzer) DeriveManifestAndLock(dir string, pr gps.ProjectRoot) (gps.Manifest, gps.Lock, error) {
+	// Ignore other tools if we find dep configuration
+	var depAnalyzer dep.Analyzer
+	if depAnalyzer.HasConfig(dir) {
+		return depAnalyzer.DeriveManifestAndLock(dir, pr)
+	}
+
+	if !a.skipTools {
+		// The assignment back to an interface prevents interface-based nil checks from failing later
+		var manifest gps.Manifest
+		var lock gps.Lock
+		im, il, err := a.importManifestAndLock(dir, pr)
+		if im != nil {
+			manifest = im
+		}
+		if il != nil {
+			lock = il
+		}
+		return manifest, lock, err
+	}
+
+	return nil, nil, nil
+}
+
+func (a *rootAnalyzer) FinalizeManifestAndLock(m *dep.Manifest, l *dep.Lock) {
+	// Iterate through the new projects in solved lock and add them to manifest
+	// if direct deps and log feedback for all the new projects.
+	for _, x := range l.Projects() {
+		pr := x.Ident().ProjectRoot
+		newProject := true
+		// Check if it's a new project, not in the old lock
+		for _, y := range a.origL.Projects() {
+			if pr == y.Ident().ProjectRoot {
+				newProject = false
+			}
+		}
+		if newProject {
+			// Check if it's in notondisk project map. These are direct deps, should
+			// be added to manifest.
+			if _, ok := a.pd.notondisk[pr]; ok {
```

Comment:
is `notondisk` still the thing that we should be checking for this, given that that ought to be specific to the GOPATH search?